Source/WebKit/NetworkProcess/cache/NetworkCache.cpp

```diff
@@ -111,20 +112,15 @@ Cache::Cache(NetworkProcess& networkProcess, const String& storageDirectory, Ref
 {
 #if ENABLE(NETWORK_CACHE_SPECULATIVE_REVALIDATION)
     if (options.contains(CacheOption::SpeculativeRevalidation)) {
-        m_lowPowerModeNotifier = makeUnique<WebCore::LowPowerModeNotifier>([weakThis = WeakPtr { *this }](bool isLowPowerModeEnabled) {
-            ASSERT(WTF::RunLoop::isMain());
-            RefPtr protectedThis = weakThis.get();
-            if (!protectedThis)
-                return;
-
-            if (isLowPowerModeEnabled)
-                protectedThis->m_speculativeLoadManager = nullptr;
-            else {
-                ASSERT(!protectedThis->m_speculativeLoadManager);
-                protectedThis->m_speculativeLoadManager = makeUnique<SpeculativeLoadManager>(*protectedThis, protectedThis->protectedStorage());
-            }
+        m_lowPowerModeNotifier = makeUnique<WebCore::LowPowerModeNotifier>([this, weakThis = WeakPtr { *this }](bool) {
+            if (RefPtr protectedThis = weakThis.get())
+                updateSpeculativeLoadManagerEnabledState();
         });
-        if (!m_lowPowerModeNotifier->isLowPowerModeEnabled())
+        m_thermalMitigationNotifier = makeUnique<WebCore::ThermalMitigationNotifier>([this, weakThis = WeakPtr { *this }](bool) {
```

Comment:
Do we have release logging when we apply (aggressive) thermal mitigations?

Reply:
Nope. I'll add some logging.